packages/package_info_plus/package_info_plus_web/lib/package_info_plus_web.dart

```diff
@@ -10,18 +10,31 @@ import 'package:package_info_plus_platform_interface/package_info_platform_inter
 ///
 /// This class implements the `package:package_info_plus` functionality for the web.
 class PackageInfoPlugin extends PackageInfoPlatform {
+  final Client _client;
+
+  /// Create plugin with http client.
+  PackageInfoPlugin(this._client);
+
   /// Registers this class as the default instance of [PackageInfoPlatform].
   static void registerWith(Registrar registrar) {
-    PackageInfoPlatform.instance = PackageInfoPlugin();
+    PackageInfoPlatform.instance = PackageInfoPlugin(Client());
+  }
+
+  /// Get version.json full url.
+  Uri versionJsonUrl(String baseUrl, int cacheBuster) {
+    final baseUri = Uri.parse(baseUrl);
+    final originPath = '${baseUri.origin}${baseUri.path}';
+    final versionJson = 'version.json?cachebuster=$cacheBuster';
+    return Uri.parse(originPath.endsWith('/')
+        ? '$originPath$versionJson'
+        : '$originPath/$versionJson');
   }
 
   @override
   Future<PackageInfoData> getAll() async {
     final cacheBuster = DateTime.now().millisecondsSinceEpoch;
-    final url = Uri.parse(
-        '${Uri.parse(window.document.baseUri!).removeFragment()}version.json?cachebuster=$cacheBuster');
-
-    final response = await get(url);
```

Comment:
here is when we use get without changing the constructor. do we need to change this? any reason?